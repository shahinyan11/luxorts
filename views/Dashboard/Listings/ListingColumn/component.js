import PropTypes from 'prop-types';
import Link from 'next/link';
import Image from 'next/image';

import ROUTES from 'constants/routes';
import { LISTING_IMAGE_PLACEHOLDER } from 'constants/listing';

import { createRoute } from 'utils/createRouteHelpers';

import FormattedOrRawMessage from 'views/shared/FormattedOrRawMessage';

const ListingColumn = ({ listing }) => {
  const title = listing.title ||
    listing.location ||
    listing.mainListingPhoto?.description || { id: 'shared.withoutTitle' };
  const listingDetailsRoute = createRoute({ pathname: ROUTES.LISTING.PATH, id: listing.id });

  return (
    <div className="d-flex align-items-center">
      <div className="mr-8 mr-sm-16">
        <Link href={listingDetailsRoute}>
          <a>
            <Image
              alt=""
              width="68"
              height="48"
              src={listing?.mainListingPhoto?.photoUrls?.medium || LISTING_IMAGE_PLACEHOLDER}
              className="table__housing-img"
            />
          </a>
        </Link>
      </div>
      <p className="mb-0 table__column-name">
        <Link href={listingDetailsRoute}>
          <a>
            <FormattedOrRawMessage message={title} />
          </a>
        </Link>
      </p>
    </div>
  );
};

ListingColumn.propTypes = {
  listing: PropTypes.shape().isRequired,
};

export default ListingColumn;
