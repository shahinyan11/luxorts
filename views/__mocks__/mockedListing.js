export default {
  id: '0645baf3-945d-4122-8a0b-5ab3e95a9b3e',
  applicationStep: 'accomodation',
  status: 'in_draft',
  userAccount: { id: '6cbf2b68-4a26-41b5-bcd5-161cc37e1420', type: 'hostAccount' },
  listingPropertyType: {
    id: 'ea19f127-bc89-4799-bdbb-346126ac3845',
    typeOfPlace: 'private_place',
    propertyType: {
      id: '51e35121-c27e-46fc-9feb-b51dab2490ee',
      name: 'repudiandae',
      description: 'description repudiandae',
    },
  },
  listingLocation: {
    id: '2b4032ec-a469-4489-b7dc-536d8f9552eb',
    country: 'United States',
    city: 'Phoenix',
    state: 'Arizona',
    street: 'West Maryland Avenue',
    apartmentNumber: '12',
    zipCode: '12345',
    longitude: '-73.97447621784085',
    latitude: '40.764468117826794',
  },
  listingDescription: null,
  listingPricing: {
    pricePerDay: 64,
  },
};
