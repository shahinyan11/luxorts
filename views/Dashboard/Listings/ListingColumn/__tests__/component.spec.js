import { shallow } from 'enzyme';

import mockedListing from 'views/__mocks__/mockedListing';

import ListingColumn from '../component';

describe('ListingColumn component', () => {
  const props = { listing: mockedListing };

  const component = shallow(<ListingColumn {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
