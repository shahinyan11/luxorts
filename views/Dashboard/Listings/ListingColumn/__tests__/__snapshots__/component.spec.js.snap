// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingColumn component matches snapshot 1`] = `
<div
  className="d-flex align-items-center"
>
  <div
    className="mr-8 mr-sm-16"
  >
    <Link
      href="/listing/0645baf3-945d-4122-8a0b-5ab3e95a9b3e"
    >
      <a>
        <Image
          alt=""
          className="table__housing-img"
          height="48"
          src="/images/housing-preview-01.png"
          width="68"
        />
      </a>
    </Link>
  </div>
  <p
    className="mb-0 table__column-name"
  >
    <Link
      href="/listing/0645baf3-945d-4122-8a0b-5ab3e95a9b3e"
    >
      <a>
        <FormattedOrRawMessage
          message={
            Object {
              "id": "shared.withoutTitle",
            }
          }
        />
      </a>
    </Link>
  </p>
</div>
`;
